### PR TITLE
EPMRPP-85619 || Create global integration for GitLab plugin

### PR DESCRIPTION
**Pull request by Vadim73i**

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -1,3 +1,3 @@
-version=1.0
-description=EPAM Report Portal. Gitlab plugin.
-pluginId = Gitlab
+version=23.3
```
Comment (AmsterGet) [inline gradle.properties:1]: All our plugins and services should follow semantic versioning.
So please use version 1.0.

File: project-properties.gradle [hunk 2]
```diff
@@ -4,7 +4,7 @@ targetCompatibility = JavaVersion.VERSION_11
 def commonScriptsUrl = 'https://raw.githubusercontent.com/reportportal/gradle-scripts/'
 
 project.ext {
-    publishRepo = "https://maven.pkg.github.com/reportportal/plugin-bts-gitlab"
+    publishRepo = "https://maven.pkg.github.com/reportportal/plugin-gitlab-epam"
```
Comment (AmsterGet) [inline project-properties.gradle:7]: What is the reason of this rename?

File: ui/README.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# Plugin template for Epam Report Portal
```
Comment (AmsterGet) [inline ui/README.md:1]: Please update title.

File: ui/README.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+Install the dependencies: `npm install`
+
+Run in dev mode:
+```bash
+npm run dev # Run webpack in dev watch mode
+npm run start # Serve built files
+```
+
+_Available only from RP v5.8_: use
```
Comment (AmsterGet) [inline ui/README.md:16]: Update required version to 23.3.

File: ui/package.json [hunk 5]
```diff
@@ -2,56 +2,83 @@
...
       "git add"
     ]
   },
   "browserslist": [
     "ie 11"
+  ],
+  "keywords": [
+    "epam",
+    "reportportal",
+    "rp"
```
Comment (AmsterGet) [inline ui/package.json:82]: We can extend it now and add 'gitlab', 'bts', 'plugin'.

File: ui/tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+      "moduleFederation/common": ["src/types/moduleFederation/common"],
+      "moduleFederation/utils": ["src/types/moduleFederation/utilsTypes"],
+      "moduleFederation/libs": ["src/types/moduleFederation/libsTypes"],
+      "moduleFederation/actions": ["src/types/moduleFederation/actionsTypes"],
+      "moduleFederation/validators": ["src/types/moduleFederation/validatorTypes"],
+      "extension/constants": ["src/constants"],
+      "utils": ["src/utils"],
+      "hooks": ["src/hooks"],
+      "extension/types": ["src/types/pluginTypes"]
+
```
Comment (AmsterGet) [inline ui/tsconfig.json:29]: What is the reason of adding all these aliases here?

File: ui/src/components/integrationSettings/integrationSettings.tsx [hunk 7]
```diff
@@ -1,21 +1,20 @@
...
+import { FC } from 'react';
 
-export const IntegrationSettings = (props) => {
+import { LABELS } from '../constants';
+
+export const IntegrationSettings: FC<IntegrationSettingsInterface> = (props) => {
   const { data, goToPreviousPage, onUpdate, isGlobal, ...extensionProps } = props;
   const {
-    lib: { React, useDispatch },
+    lib: { useDispatch },
```
Comment (AmsterGet) [inline ui/src/components/integrationSettings/integrationSettings.tsx:9]: Please import `useDispatch` from `react-redux` lib.

File: ui/src/components/integrationFormFields/integrationFormFields.tsx [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+        validate={btsProjectId}
+        disabled={disabled}
+        isRequired
+      >
+        <FieldErrorHint provideHint={false}>
+          <FieldText maxLength={55} defaultWidth={false} />
+        </FieldErrorHint>
+      </FieldElement>
+      <FieldElement
+        name="apiToken"
```
Comment (AmsterGet) [inline ui/src/components/integrationFormFields/integrationFormFields.tsx:51]: Let's use field names as constants, especially if they used several times.

**Review by AmsterGet — CHANGES_REQUESTED** [review]

**Review by AmsterGet — CHANGES_REQUESTED** [review]

File: ui/src/components/integrationFormFields/integrationFormFields.tsx [hunk 9]
```diff
@@ -0,0 +1,68 @@
+import { IntegrationFormFieldsInterface } from 'moduleFederation/common';
+import React, { FC, useEffect } from 'react';
```
Comment (Bam6ycha) [inline ui/src/components/integrationFormFields/integrationFormFields.tsx:2]: Actually we do not need to import React here, it imports automatically in compilation time

**Review by Bam6ycha — CHANGES_REQUESTED** [review]

**Comment by Bam6ycha** [comment]
Also, I have a problem with .gitignore, it does not include node_modules

**Review by Bam6ycha — APPROVED** [review]

**Comment by Vadim73i** [comment]
> Also, I have a problem with .gitignore, it does not include node_modules

ui/.gitignore includes "node_modules/"

**Pull request merged**